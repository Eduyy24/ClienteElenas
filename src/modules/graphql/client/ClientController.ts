import {useMutation, useQuery} from '@apollo/client';
import { useEffect, useState } from 'react';
import { LOGIN_GQL } from './gql/mutations';
import { CLIENTS_SEARCH_GQL } from './gql/queries';
import ClientModel from './model/ClientModel';

export const getLoginMutation = () => useMutation(LOGIN_GQL);

/**
 * Ejecuta el Query para @var {CLIENTS_SEARCH_GQL}
 * @returns {ClientModel[]}
 */
export const getClients = (): ClientModel[] => {
  const clientsData = useQuery(CLIENTS_SEARCH_GQL)
  const [clients, setClients] = useState(Array<ClientModel>(0))
  
  useEffect(() => {
    if (clientsData.data) {
      setClients(clientsData.data.clientsSearch.results);
    }
  }, [clientsData.data]);

  return clients;
}