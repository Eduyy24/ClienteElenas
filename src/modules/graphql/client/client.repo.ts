import {useMutation} from '@apollo/client';
import { LOGIN_GQL } from './gql/mutations';

export const getLoginMutation = () => useMutation(LOGIN_GQL);