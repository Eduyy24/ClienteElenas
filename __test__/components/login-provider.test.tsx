import React from 'react';
import { render, waitFor, act } from '@testing-library/react-native'
import LoginProvider from '../../src/components/login-provider';
import { MockedProvider } from '@apollo/client/testing';
import LoginMutationMock from '../../__mocks__/login-mutation.mock';

describe('<LoginProvider />', () => {
  test('Ui del componente cuando esta cargando', () => {
    const component = 
      <MockedProvider mocks={LoginMutationMock} addTypename={false}>
        <LoginProvider />
      </MockedProvider>
    const wraper = render(component)
    const loading = wraper.queryByText('Cargando...')
    expect(loading).toBeTruthy();
    expect(wraper.toJSON()).toMatchSnapshot();
  })

  // test('Ui del componente cuando termina la carga',async () => {
  //   const component = 
  //     <MockedProvider mocks={LoginMutationMock} addTypename={false}>
  //       <LoginProvider />
  //     </MockedProvider>
    
  //   let wraper = render(component)
    
  //   //await waitFor(() => new Promise(resolve => setTimeout(resolve, 0)))
  //   act(() => jest.advanceTimersByTime(1000))

  //   const loading = wraper.queryByText('Clientes Elenas')
  //   expect(loading).toBeFalsy();
  //   //expect(wraper.toJSON()).toMatchSnapshot();
  // })
  
})