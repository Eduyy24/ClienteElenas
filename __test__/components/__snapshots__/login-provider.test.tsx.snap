// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginProvider /> Ui del componente cuando esta cargando 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Cargando...
  </Text>
</View>
`;

exports[`<LoginProvider /> Ui del componente cuando termina la carga 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Cargando...
  </Text>
</View>
`;
